docs/core/tools/dotnet-nuget-add-source.md

```diff
@@ -0,0 +1,88 @@
+---
+title: dotnet nuget add source command
+description: The `dotnet nuget add source` command adds a new package source to your NuGet configuration files. 
+author: nugetClient
+ms.date: 03/09/2020
+---
+# dotnet nuget add source
+
+**This article applies to:** ✔️ .NET Core 3.1.200 SDK and later versions
+
+## Name
+
+`dotnet nuget add source` - Add a NuGet source.
+
+## Synopsis
+
+```dotnetcli
+`dotnet nuget add source PACKAGESOURCEPATH [--name] [--username] [--password] [--store-password-in-clear-text] [--valid-authentication-types] [--configfile]`
+`dotnet nuget add source [-h|--help]`
+```
+
+## Description
+
+The `dotnet nuget add source` command adds a new package source to your NuGet configuration files. 
+
+## Arguments
+
+- **`PACKAGESOURCEPATH`**
+
+  Path to the package(s) source.
+
+## Options
+
+- **`-n|--name`**
+
+  Name of the source.
+
+- **`-u|--username`**
+
+  Username to be used when connecting to an authenticated source.
+
+- **`-p|--password`**
+
+  Password to be used when connecting to an authenticated source.
+
+- **`--store-password-in-clear-text`**
+
+  Enables storing portable package source credentials by disabling password encryption.
+
+- **`--valid-authentication-types`**
+
+  Comma-separated list of valid authentication types for this source. By default, all authentication types are valid. Example: basic,negotiate
```

Comment:
looking into this.

Reply:
Spoke with Rob offline -- yes and no. The framework sends us a string when it's trying to get credentials to add to a request, and we compare against this setting. Technically the set of strings that are possible depends what providers are configured, but I believe the configuration has to be done in machine.config (on netfx anyway) so probably nobody reconfigures that.  It looks like the default providers provide `negotiate`, `kerberos`, `ntlm`, `digest` and `basic`, but in practice only `basic` and *maybe* `ntlm` or `negotiate` are going to be useful to users.

As background, the original goal of the feature was to *exclude* NTLM auth. Azure DevOps Server supports auth using PATs and other non-Windows credentials. Some identities in the system don't even exist in AD, the identities used by Pipelines for example. Without this setting, NuGet would try to use these non-Windows credentials with NTLM, fail, and not fall back to Basic. With the setting set to `basic`, when the framework asks for NTLM (or any other kind of) credentials, NuGet just responds "nope, no credentials here, try something else"

Reply:
Much of this seems like it would be useful to include in the doc, unless it exists somewhere else that we can link to.

Reply:
I think most of this information would actually just be confusing to most users. Something scenario-focused would probably be clearer.

"Set this to `basic` if the server advertises NTLM or Negotiate and your credentials must be sent using the Basic mechanism, for instance when using a PAT with on-premises Azure DevOps Server"?

Reply:
OK --

> but in practice only basic and maybe ntlm or negotiate are going to be useful to users.

Can we also then specify the criteria for setting to ntlm or negotiate?

Reply:
I don't think that's necessary. With the default configuration, `negotiate` is tried first, and that should support NTLM, and `ntlm` is tried before `basic`, so I can't think of a scenario where setting it to `ntlm` or `negotiate` would meaningfully change the behavior

Reply:
Aside from this scenario with `basic`, if you have enough information to know you need to set this option, you will know what value to set it to just given the list, because you will have been pulling out your hair looking at Fiddler traces watching NuGet try to use the wrong authentication mechanism

Reply:
thanks jonathan. ended up using your proposed wording. fixed in latest commit.